### PR TITLE
ccache: update to 4.0

### PR DESCRIPTION
**Pull request by pullmoll**
This release changes from using autoconf to cmake which is why
we need to remove the bootstrap=yes (N.B.: cmake is not bootstrap=yes).

I think we cannot turn `cmake` into a `bootstrap=yes` package.
Should we create a (legacy) chroot-ccache with a < 4.0 version?

There seems to be no way to build `ccache` >= 4.0 with just autoconf.

### CODE REVIEW

**Comment by sgn** [comment]
Alternately, we can package a chroot-cmake with vendored libraries for bootstrap.

**Comment by pullmoll** [comment]
@sgn That would be better because the cache data format also changed.
Using the same cache directory for a `chroot-ccache-3.x` and `ccache` >= 4.0 would work but not re-use the cache data.
If you can come up with a `chroot-cmake` that would be great.

**Comment by pullmoll** [comment]
FYI working on a `chroot-cmake` package. It builds but I need to strip it down to the required stuff.
I wonder if we should make `expat` a `bootstrap=yes` package.
`cmake` depends on it and it's simple enough, i.e. no dependencies.

**Comment by sgn** [comment]
I also have a working `chroot-cmake`, but with vendored `expat`. Not sure about bootstrap=yes for expat.

**Comment by pullmoll** [comment]
It works with the `--without-system-expat` configure option. Still, using system expat would make sense.
I don't think it's a problem to turn `expat` into `bootstrap=yes`. It's tiny compared to cmake.
Need to strip down chroot-cmake to what we really need.

**Comment by pullmoll** [comment]
Oops.. there's a wrong change to `srcpkgs/chroot-cmake/template` in a later commit. Need to sort that out but have to go now.
It might make sense to try and remove unused compilers from the build, if that's possible. Or otherwise strip out what we definitely don't need.
Will take a look into that when I'm back this evening.

File: srcpkgs/chroot-cmake/template [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ --system-zlib
+ --system-bzip2
+ --system-liblzma
+ --no-system-nghttp2
+ --system-zstd
+ --system-libarchive
+ --no-system-librhash
+ --no-system-libuv
+ ${XBPS_MAKEJOBS:+--parallel=$XBPS_MAKEJOBS}"
+makedepends="libarchive-devel ncurses-devel"
```
Comment (sgn) [inline srcpkgs/chroot-cmake/template:23]: I don't think it need `ncurses-devel`

**Comment by pullmoll** [comment]
Now let's see if it builds w/o ncurses and with system expat...

**Comment by sgn** [comment]
@pullmoll I've draft another approach for `chroot-cmake`. Please take a look in my approach. https://github.com/void-linux/void-packages/pull/25753

**Comment by pullmoll** [comment]
Closing in favor of #25753

**Pull request closed without merging**